### PR TITLE
New test folder structure for run_task unit tests

### PR DESCRIPTION
**Pull request by nathanjmcdougall**
This is just a starting point for how to structure the test suite. I wanted to run it past you before I do too much more.

Probably makes sense to wait until #7 is merged.

### CODE REVIEW

File: tests/labtech/test_init__.py
Comment (ben-denham) [inline tests/labtech/test_init__.py:1]: Should this be a `__init__.py` file so that the tests are discovered?

**Review by ben-denham — COMMENTED** [review]
This PR looks great thanks, just one question:

File: tests/labtech/test_init__.py
Comment (nathanjmcdougall) [reply tests/labtech/test_init__.py:1]: I think there should be an empty `__init__.py` _in addition_ to this one, yes, good catch.

At some point I would like to set up a linter rule which checks for missing `__init__.py` files.

Comment (nathanjmcdougall) [reply tests/labtech/test_init__.py:1]: Actioned.

Comment (ben-denham) [reply tests/labtech/test_init__.py:1]: Great, thanks. I'm still not quite sure what this `test_init__.py` is for, should it have the version number test in it from `test_labtech.py`? Should it be `test__init__.py` instead?

File: tests/labtech/test_lab.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class TestLab:
+    class TestRunTasks:
+        def test_no_tasks(self, lab: Lab) -> None:
+            lab.run_tasks(tasks=[])
+
+        def test_simple_task(self, lab: Lab) -> None:
+            lab.run_tasks(tasks=[_SimpleTask(a=1)])
+
+        def test_noparam_task(self, lab: Lab) -> None:
+            lab.run_tasks(tasks=[_NoParamTask()])
```
Comment (ben-denham) [inline tests/labtech/test_lab.py:22]: Any reason to not check the results while we're running these tests?

```suggestion
        def test_no_tasks(self, lab: Lab) -> None:
            results = lab.run_tasks(tasks=[])
            assert results == {}

        def test_simple_task(self, lab: Lab) -> None:
            results = lab.run_tasks(tasks=[_SimpleTask(a=1)])
            assert results == {_SimpleTask(a=1): 1}

        def test_noparam_task(self, lab: Lab) -> None:
            results = lab.run_tasks(tasks=[_NoParamTask()])
            assert results == {_NoParamTask(): 1}
```

File: tests/labtech/test_init__.py
Comment (nathanjmcdougall) [reply tests/labtech/test_init__.py:1]: My motivation is that for unit tests, I try and keep the folder structure of the `tests` folder to match the package folder.
To test things defined in an `__init__.py` file, I want a corresponding test file, name doesn't matter. `test__init__.py` is arguably better but the double underscore in the middle of the variable name is a bit of a gotcha. Happy either way, as long as we are consistent.

The tests in this file are just the same as the ones in the (deleted) file `test_version.py`

File: tests/labtech/test_lab.py (hunk 1 above)
Comment (nathanjmcdougall) [reply tests/labtech/test_lab.py]: Fair suggestion - for me the point of those tests was just checking that the function runs without error - no harm in checking output too. I have committed your suggestion.

File: tests/labtech/test_init__.py
Comment (ben-denham) [reply tests/labtech/test_init__.py:1]: Ah yep, I see now, I wasn't reading carefully enough to see that it was a file rename and not just the creation of an empty file.

**Pull request merged**